superset/sql_parse.py

```diff
@@ -57,10 +58,30 @@ def _extract_limit_from_query(statement: TokenList) -> Optional[int]:
     return None
 
 
+@dataclass(eq=True, frozen=True)
+class Table:  # pylint: disable=too-few-public-methods
+    """
+    A fully specified SQL table conforming to [[catalog.]schema.]table.
+    """
+
+    table: str
+    schema: Optional[str] = None
+    catalog: Optional[str] = None
+
+    def __str__(self) -> str:
+        """
+        Return the fully specified SQL table name.
+        """
+
+        return ".".join(
+            part for part in [self.catalog, self.schema, self.table] if part
+        )
```

Comment:
This is the perfect opportunity to start using escaped fully qualified names to simplify downstream logic. I propose we add `urllib.parse.quote(part, safe="")` here to make sure tables/schemas/catalogs with periods and other special characters are properly escaped.